### DISCUSSION TITLE
Trailing zero factors in rational functions

### DISCUSSION THREAD

**Discussion by ghalimi** [stdlib-js/stdlib#377] [Q&A] [answered]
What is the rationale for including trailing zero factors in rational functions such as [`s2`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/erfinv/lib/rational_p3q3.js#L53), which is used to implement  [erfinv](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/erfinv)? In this function, the last three factors are equal to zero. Is it to facilitate code reading (at the expense of a minuscule degradation in performance)? Is it because the code was generated from metadata and no routine was implemented to get rid of these trailing zero factors? Or is there another reason for this?

**Comment by kgryte** [comment, marked as answer]
The coefficients originate from a [script](https://github.com/stdlib-js/stdlib/blob/4aa3d37dc6d7dde15c001382716abe62a7450742/lib/node_modules/%40stdlib/math/base/special/erfinv/scripts/evalrational.js). Those coefficients are subject to change. Modern JITs can leverage constant folding to avoid unnecessary computation.

**Reply by ghalimi** [reply]
That's what I was expecting. Thanks!
